src/deque.jl

```diff
@@ -271,3 +271,20 @@ function shift!{T}(q::Deque{T})  # pop front
     q.len -= 1
     x
 end
+
+const _deque_hashseed = UInt === UInt64 ? 0x950aa17a3246be82 : 0x4f26f881
+function hash(x::Deque)
+    h = _deque_hashseed
+    for (i, x) in enumerate(x)
+        h += i * hash(x)
+    end
+    h
+end
```

Comment:
Sorry, one more comment: this should be changed to the 2-argument version of `hash` (I just discovered this).

```julia
help?> hash
search: hash haskey skipchars blas_set_num_threads searchsortedlast with_bigfloat_precision

  hash(x[, h::UInt])

  Compute an integer hash code such that isequal(x,y) implies hash(x)==hash(y). The optional second argument h is a hash code to be mixed with the result.

  New types should implement the 2-argument form, typically by calling the 2-argument hash method recursively in order to mix hashes of the contents with each other (and with h). Typically, any type that implements hash should also
  implement its own == (hence isequal) to guarantee the property mentioned above.
```

See https://github.com/JuliaLang/julia/blob/master/base/abstractarray.jl#L1415 for a (slightly complicated) example.